### PR TITLE
DH-524 Save event

### PR DESCRIPTION
**Pull request by yeahfro**
This is the final (hopefully!) PR for the saving of an event.

Included in this PR:
- Final changes to event transformation, based upon validation errors received from API
- Including errors when building the form
- Adding a basic view for viewing the event after save.

# Form with errors
screen shot 2017-09-19 at 11 42 46

# On successful save
screen shot 2017-09-19 at 11 43 38

### CODE REVIEW

File: src/apps/events/controllers/edit.js [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
     const eventForm = eventFormConfig(advisers)
     const emptyDate = { year: '', month: '', day: '' }
-    const body = Object.assign(req.body, {
+    const body = assign({}, req.body, {
       start_date: emptyDate,
       end_date: emptyDate,
       lead_team: get(req, 'session.user.dit_team.id', null),
     })
-    const eventFormWithState = buildFormWithState(eventForm, body)
+    const eventFormWithStateAndErrors =
```
Comment (feedmypixel) [inline src/apps/events/controllers/edit.js]: is the fact its with state and errors implicit? could this not just be `eventForm`? 
Is the name  `eventFormWithStateAndErrors ` a symptom of iteratively building this?

File: src/apps/events/router.js [hunk 2]
```diff
@@ -1,9 +1,22 @@
...
+    edit.renderEventPage,
+  )
+  .post(
+    detailsFormMiddleware.handleFormPost,
+    edit.postHandler,
+    edit.renderEventPage,
+  )
+
+router.route('/:id/details')
+  .get(details.renderPage)
```
Comment (feedmypixel) [inline src/apps/events/router.js]: If its just 1 verb we tend to do: `router.get('/:id/details', details.renderPage)`

File: src/apps/events/services/formatting.js [hunk 3]
```diff
@@ -47,6 +47,8 @@ function transformToApi (body) {
   setDate('start_date')
   setDate('end_date')
 
+  formatted.teams = union(formatted.teams, [ formatted.lead_team ])
+  formatted.service = '1783ae93-b78f-e611-8c55-e4115bed50dc'
```
Comment (feedmypixel) [inline src/apps/events/services/formatting.js]: 👮 🚨 is this meant to be here?

File: src/apps/events/services/formatting.js [hunk 4]
```diff
@@ -47,6 +47,8 @@ function transformToApi (body) {
   setDate('start_date')
   setDate('end_date')
 
+  formatted.teams = union(formatted.teams, [ formatted.lead_team ])
```
Comment (feedmypixel) [inline src/apps/events/services/formatting.js]: is there a need for the resulting array to contain unique values? could you just push `formatted.lead_team` onto `formatted.teams`?

```
formatted.teams.push(formatted.lead_team)
```

File: test/unit/apps/events/controllers/details.test.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+describe('Event details controller', () => {
+  beforeEach(() => {
+    this.controller = require('~/src/apps/events/controllers/details')
+
+    this.sandbox = sinon.sandbox.create()
+
+    this.req = {
```
Comment (feedmypixel) [inline test/unit/apps/events/controllers/details.test.js]: can go on 1 line

File: src/apps/events/controllers/edit.js (hunk 1 above)
Comment (yeahfro) [reply src/apps/events/controllers/edit.js]: I will fix this naming as it has evolved into something cumbersome.

File: src/apps/events/services/formatting.js (hunk 4 above)
Comment (yeahfro) [reply src/apps/events/services/formatting.js]: Fair enough, I will simplify this.

File: src/apps/events/services/formatting.js (hunk 3 above)
Comment (yeahfro) [reply src/apps/events/services/formatting.js]: This is being hardcoded for now as there is a subsequent JIRA ticket that will add this field. It is required for now to get passed API validation.

**Pull request merged**